### PR TITLE
Fix mypy errors in airflow/operators

### PR DESCRIPTION
**Pull request by subkanthi**
related: #19891 
Fixes mypy errors in airflow/operators

airflow/operators/python.py:179: error: Incompatible types in assignment (expression has type "Mapping[str, Any]", variable has type "Dict[Any, Any]")
            self.op_kwargs = determine_kwargs(self.python_callable, self.op_args, context)
                             ^
airflow/operators/bash.py:146: error: Incompatible default for argument "cwd" (default has type "None", argument has type "str")
            cwd: str = None,
                       ^
airflow/operators/trigger_dagrun.py:173: error: Item "None" of "Optional[DagRun]" has no attribute "execution_date"
            ti.xcom_push(key=XCOM_EXECUTION_DATE_ISO, value=dag_run.execution_date.isoformat())
                                                            ^
airflow/operators/trigger_dagrun.py:174: error: Item "None" of "Optional[DagRun]" has no attribute "run_id"
            ti.xcom_push(key=XCOM_RUN_ID, value=dag_run.run_id)
                                                ^
airflow/operators/trigger_dagrun.py:182: error: Item "None" of "Optional[DagRun]" has no attribute "execution_date"
                        dag_run.execution_date,
                        ^
airflow/operators/trigger_dagrun.py:187: error: Item "None" of "Optional[DagRun]" has no attribute "refresh_from_db"
                    dag_run.refresh_from_db()
                    ^
airflow/operators/trigger_dagrun.py:188: error: Item "None" of "Optional[DagRun]" has no attribute "state"
                    state = dag_run.state

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: airflow/operators/trigger_dagrun.py
Comment (subkanthi) [inline airflow/operators/trigger_dagrun.py:170]: Need to validate if this is OK to do

Comment (potiuk) [reply airflow/operators/trigger_dagrun.py:170]: I believe  it's good, though error message could be more descriptive. "Could not find dag_run to trigger" ?.  What do others think ?

**Review by potiuk — APPROVED** [review]
LGTM. But needs other committer's approval.

**Pull request merged**
